tests/specialization_constants/specialization_constants_separate_unit.cpp

```diff
@@ -37,25 +37,25 @@ bool check_kernel_handler_by_value_external(sycl::kernel_handler h,
   SYCL_EXTERNAL bool check_kernel_handler_by_reference_external_handler(       \
       sycl::kernel_handler &h, TYPE expected) {                                \
     return check_kernel_handler_by_reference_external<                         \
-        TYPE, by_reference_via_handler>(h, expected);                          \
+        TYPE, (int)test_cases_external::by_reference_via_handler>(h, expected);\
   }                                                                            \
                                                                                \
   SYCL_EXTERNAL bool check_kernel_handler_by_value_external_handler(           \
       sycl::kernel_handler h, TYPE expected) {                                 \
-    return check_kernel_handler_by_value_external<TYPE, by_value_via_handler>( \
-        h, expected);                                                          \
+    return check_kernel_handler_by_value_external<                             \
+        TYPE, (int)test_cases_external::by_value_via_handler>(h, expected);    \
   }                                                                            \
                                                                                \
   SYCL_EXTERNAL bool check_kernel_handler_by_reference_external_bundle(        \
       sycl::kernel_handler &h, TYPE expected) {                                \
     return check_kernel_handler_by_reference_external<                         \
-        TYPE, by_reference_via_bundle>(h, expected);                           \
+        TYPE, (int)test_cases_external::by_reference_via_bundle>(h, expected); \
   }                                                                            \
                                                                                \
   SYCL_EXTERNAL bool check_kernel_handler_by_value_external_bundle(            \
       sycl::kernel_handler h, TYPE expected) {                                 \
-    return check_kernel_handler_by_value_external<TYPE, by_value_via_bundle>(  \
-        h, expected);                                                          \
+    return check_kernel_handler_by_value_external<                             \
+        TYPE, (int)test_cases_external::by_value_via_bundle>(h, expected);     \
```

Comment:
Naïve question: it possible to improve the test to avoid all these cast?
Such as changing `check_kernel_handler_by_reference_external` to accept directly a `test_cases_external::by_value_via_bundle`?

Reply:
Improved a little bit, thanks. But I left `template <typename T, int case_num> ... spec_const_external(...)` with `int` for consistency with all other spec const tests, if you don't mind